rio/srcrobot/robot_container.py

```diff
@@ -159,16 +158,22 @@ def configureButtonBindings(self):
         self.faceBack.onTrue(TurnInPlace(self.s_Swerve, lambda: (Rotation2d.fromDegrees(0)), translation, strafe, rotation, robotcentric))
         self.faceLeft.onTrue(TurnInPlace(self.s_Swerve, lambda: (Rotation2d.fromDegrees(90)), translation, strafe, rotation, robotcentric))
         self.faceRight.onTrue(TurnInPlace(self.s_Swerve, lambda: (Rotation2d.fromDegrees(-90)), translation, strafe, rotation, robotcentric))
-        
 
         #Intake Buttons
-        self.intakeOn.onTrue(self.s_Intake.setIntakeSpeed(Constants.IntakeConstants.kIntakeSpeed).alongWith(self.s_Indexer.indexerIntake()))
-        self.intakeOff.onTrue(self.s_Intake.stopIntake().alongWith(self.s_Indexer.stopIndexer()))
+        self.s_Intake.setDefaultCommand(self.s_Intake.stopIntake())
+        self.s_Indexer.setDefaultCommand(self.s_Indexer.stopIndexer())
+        self.intake.whileTrue(self.s_Intake.intake().alongWith(self.s_Indexer.indexerIntake()))
+        self.intakeReverse.whileTrue(self.s_Intake.outtake().alongWith(self.s_Indexer.indexerOuttake()))
 
         #Shooter Buttons
-        self.shooterOff.onTrue(self.s_Shooter.stop().alongWith(self.s_Indexer.stopIndexer()))
-        self.shoot.onTrue(self.s_Shooter.shoot().andThen(WaitCommand(2.0)).andThen(self.s_Indexer.indexerShoot()))
-        self.reverse.onTrue(self.s_Shooter.shootReverse().alongWith(self.s_Indexer.indexerIntake()))
+        self.shooterRev.onTrue(self.s_Shooter.shoot())
+        self.shoot.onTrue(
+            ConditionalCommand(
+                self.s_Indexer.indexerShoot(),
+                self.s_Indexer.stopIndexer(),
+                self.s_Shooter.isShooterReady
+            )
+        )
```

Comment:
This may work but there are cleaner ways to do this. Any data source that provides (e.g. a boolean returning function), or one that can be converted to a boolean (e.g. a float and a threshold) can be used to trigger a command, not just buttons. Also triggers can be combined with boolean operators to form a composite trigger. So, this could be something like (I may have some syntax off but the concept is right):

    shooterReadyTrigger = Trigger(lambda: self.s_Shooter.isShooterReady())
    self.shoot.and(shooterReadyTrigger).onTrue(self.s_Indexer.indexerShoot())